diff -aurBb review.phtml reviewcopy.phtml
--- review.phtml	2015-07-13 16:40:33.000000000 +0300
+++ reviewcopy.phtml	2015-07-13 17:04:05.000000000 +0300
@@ -27,6 +27,7 @@
 
 $billingBlock = $this->getChild('billing')->setFieldNamePrefix('billing')->setHideEmailAddress(true);
 $shippingAddress = $this->getShippingAddress();
+$cfmHelper = Mage::helper('aitcheckoutfields');
 ?>
 <div class="page-title">
     <h1><?php echo $this->__('Review Order') ?></h1>
@@ -60,21 +61,39 @@
     <div class="info-set col2-set">
         <div class="col-1" id="billing-address">
             <h2 class="legend"><?php echo $this->__('Billing Address') ?></h2>
+
+            <?php echo $cfmHelper->getStepFieldsCode('billing', 1, 'onepage', 'aitpaypalexpress'); ?>
+            <?php echo $cfmHelper->getStepFieldsCode('payment', 1, 'onepage', 'aitpaypalexpress'); ?>
+
             <?php if ($shippingAddress): ?>
                 <?php echo $billingBlock->setShowAsShippingCheckbox(true)->toHtml(); ?>
             <?php else: ?>
                 <?php echo $billingBlock->toHtml(); ?>
             <?php endif; ?>
+
+            <?php echo $cfmHelper->getStepFieldsCode('billing', 2, 'onepage', 'aitpaypalexpress'); ?>
+            <?php echo $cfmHelper->getStepFieldsCode('payment', 2, 'onepage', 'aitpaypalexpress'); ?>
+
         </div>
     <?php if ($shippingAddress): ?>
         <div class="col-2" id="shipping-address">
             <h2 class="legend"><?php echo $this->__('Shipping Address') ?></h2>
+
+            <?php echo $cfmHelper->getStepFieldsCode('shippinfo', 1, 'onepage', 'aitpaypalexpress'); ?>
+        
+            <?php echo $this->getChild('shipping')->setFieldNamePrefix('shipping')->setHideEmailAddress(true)->toHtml(); ?>
+        
+            <?php echo $cfmHelper->getStepFieldsCode('shippinfo', 2, 'onepage', 'aitpaypalexpress'); ?>
+
             <?php echo $this->getChild('shipping')->setFieldNamePrefix('shipping')->setHideEmailAddress(true)->toHtml(); ?>
         </div>
     </div>

     <div class="info-set col2-set">
         <div class="col-2">
+
+            <?php echo $cfmHelper->getStepFieldsCode('shippmethod', 1, 'onepage', 'aitpaypalexpress'); ?>
+
             <div class="box paypal-shipping-method">
                 <div class="box-title">
                     <h3><?php echo $this->__('Shipping Method') ?></h3>
@@ -83,15 +102,24 @@
                     <?php echo $this->getChild('shipping_method')->toHtml(); ?>
                 </div>
             </div>
+
+            <?php echo $cfmHelper->getStepFieldsCode('shippmethod', 2, 'onepage', 'aitpaypalexpress'); ?>
+
         </div>
     <?php endif; ?>
     </div>
 
     <div class="info-set">
         <h2 class="legend"><?php echo $this->__('Items in Your Shopping Cart') ?><span class="separator"> | </span><a href="<?php echo $this->getUrl('checkout/cart') ?>"><?php echo $this->__('Edit Shopping Cart') ?></a></h2>
+        
+        <?php echo  $cfmHelper->getStepFieldsCode('review', 1, 'onepage', 'aitpaypalexpress'); ?>
+
         <div id="details-reload">
             <?php echo $this->getChildHtml('details') ?>
         </div>
+
+        <?php echo  $cfmHelper->getStepFieldsCode('review', 2, 'onepage', 'aitpaypalexpress'); ?>
+
     </div>
         <?php echo $this->getChildHtml('agreements'); ?>
         <div class="buttons-set buttons-set-order" id="review-buttons-container">
@@ -114,12 +142,21 @@
 
 <?php if ($this->getUseAjax()):?>
     OrderReviewController.prototype._submitOrder = function() {
+
+        if(!this.validate()) {
+            return false;
+        }
+
         if (this._canSubmitOrder) {
             if (this._pleaseWait) {
                 this._pleaseWait.show();
             }
             new Ajax.Request(this.form.action, {
+                <?php /* AITCFM_COMMENT_START
                 parameters: {isAjax: 1, method: 'POST'},
+                AITCFM_COMMENT_END */ ?>
+                mathod: 'post',
+                parameters: 'isAjax=1' + ((serializedCfmForm = $('order_review_form').serialize()) ? ('&' + serializedCfmForm) : '' ),
                 onSuccess: function(transport) {
                     try{
                         response = eval('(' + transport.responseText + ')');
@@ -155,7 +192,60 @@
             });
         }
     }
+
+<?php else: ?>
+
+OrderReviewController.addMethods({
+    _submitOrder : function()
+    {
+        if(!this.validate()) {
+            return false;
+        }
+        if ($$('.aitcfm-fieldset').length) {
+            if ($('aitcfm-fieldset-tmp')) {
+                $('aitcfm-fieldset-tmp').remove();
+            }
+            var ait_fieldset_tmp = new Element('div', {id : 'aitcfm-fieldset-tmp', style: 'display:none !important;' });
+
+            $$('.aitcfm-fieldset input, .aitcfm-fieldset select, .aitcfm-fieldset textarea').each(function(el) {
+                ait_fieldset_tmp.insert(el.clone(true));
+            });
+            this.form.insert(ait_fieldset_tmp);
+        }
+
+        if (this._canSubmitOrder) {
+            this.form.submit();
+            this._updateOrderSubmit(true);
+            if (this._pleaseWait) {
+                this._pleaseWait.show();
+            }
+        }
+    }
+});
+
 <?php endif ?>
+
+OrderReviewController.addMethods({
+    validate: function()
+    {
+        if ($$('.aitcfm-fieldset').length) {
+            var ait_validate_result = true;
+            $$('.aitcfm-fieldset input, .aitcfm-fieldset select, .aitcfm-fieldset textarea').each(function(el) {
+                if (!Validation.validate(el)) {
+                    ait_validate_result = false;
+                }
+            });
+            if (ait_validate_result === false) {
+                return ait_validate_result;
+            }
+        }
+        if( typeof(this.validator) == 'undefined') {
+            this.validator = new Validation(this.form);
+        }
+        return this.validator.validate();   
+    }
+});
+
 PayPalExpressAjax = new OrderReviewController($('order_review_form'), $('review_button'),
     'shipping_method', null, 'details-reload'
 );
