diff -aurBb /home/httpd/htdocs/users/kaygorodov/discounts/var/ait_patch_orig/compare64812809962704c5a73ae09d42 /home/httpd/htdocs/users/kaygorodov/discounts/var/ait_patch_modif/compare64812809962704c5a73ae09d42
--- /home/httpd/htdocs/users/kaygorodov/discounts/var/ait_patch_orig/compare64812809962704c5a73ae09d42	2010-08-05 11:17:50 +0300
+++ /home/httpd/htdocs/users/kaygorodov/discounts/var/ait_patch_modif/compare64812809962704c5a73ae09d42	2010-08-05 11:17:50 +0300
@@ -53,6 +53,32 @@
         <?php echo $this->getPaymentInfoHtml() ?>
     </div>
 </div>
+
+
+
+<!--     START AITOC CHECKOUT ATTRIBUTES     -->
+
+<?php if ($aCustomAtrrList = $this->getOrderCustomData()) : ?>
+    <!--Account Information-->
+<br />
+<div class="col2-set">
+        <h4><?php echo Mage::getStoreConfig('aitcheckoutfields/common_settings/aitcheckoutfields_additionalblock_label', $this->getStoreId()); ?></h4>
+                <table cellspacing="0" class="print" width="100%">
+                
+    <?php foreach ($aCustomAtrrList as $aItem): ?>
+                <tr>
+                    <td class="label" width="30%"><label><?php echo $aItem['label']; ?></label></td>
+                    <td><?php echo $aItem['value']; ?></td>
+                </tr>
+    <?php endforeach; ?>
+                
+                </table>
+</div>
+
+<?php endif; ?>
+
+<!--     FINISH AITOC CHECKOUT ATTRIBUTES     -->
+
 <h2><?php echo $this->__('Items Ordered') ?></h2>
 <table class="data-table" id="my-orders-table">
     <col />
