diff -aurBb /home/httpd/htdocs_p5m/1610au1/var/ait_patch_orig/compare41313285760534f30763578233 /home/httpd/htdocs_p5m/1610au1/var/ait_patch_modif/compare41313285760534f30763578233
--- /home/httpd/htdocs_p5m/1610au1/var/ait_patch_orig/compare41313285760534f30763578233	2012-02-07 02:54:13.000000000 +0200
+++ /home/httpd/htdocs_p5m/1610au1/var/ait_patch_modif/compare41313285760534f30763578233	2012-02-07 02:54:13.000000000 +0200
@@ -1,4 +1,8 @@
-<?php
+<?php /* !!!ATTENTION!!! PLEASE DO NOT MODIFY THE FILE! 
+Copy it preserving its path from the var/ait_path folder to the
+app folder. i.e. in var/ait_path folder the file is located in folder 1, 
+then in the app folder you also need to make folder 1 and put the file in it.
+*/ ?><?php
/**
 * Magento
 *
@@ -87,7 +91,10 @@
                    </div>
                    <div class="product-secondary">
                        <?php if(!$_product->canConfigure() && $_product->isSaleable()): ?>
+                        <?php /* ?>
                         <p class="action"><button type="button" title="<?php echo $this->__('Add to Cart') ?>" class="button btn-cart" onclick="setLocation('<?php echo $this->getAddToCartUrl($_product) ?>')"><span><span><?php echo $this->__('Add to Cart') ?></span></span></button></p>
+                        <?php */ ?>
+                        <p class="action"><button type="button" title="<?php echo $this->__('Add to Cart') ?>" class="button btn-cart" onclick="aitunitsForm_productQty.send('<?php echo $this->getAddToCartUrl($_product) ?>')"><span><span><?php echo $this->__('Add to Cart') ?></span></span></button></p>
                     <?php elseif($_product->getStockItem() && $_product->getStockItem()->getIsInStock()): ?>
                            <p class="action"><a title="<?php echo $this->__('View Details') ?>" class="button" href="<?php echo $_product->getProductUrl() ?>"><?php echo $this->__('View Details') ?></a></p>
                        <?php else: ?>
@@ -95,7 +102,10 @@
                     <?php endif; ?>
                        <ul class="add-to-links">
                            <?php if ($this->helper('wishlist')->isAllow()) : ?>
+                            <?php /* ?>
                             <li><a href="<?php echo $this->helper('wishlist')->getAddUrl($_product) ?>" class="link-wishlist"><?php echo $this->__('Add to Wishlist') ?></a></li>
+                            <?php */ ?>
+                            <li><a class="link-wishlist" onclick="aitunitsForm_productQty.send('<?php echo $this->helper('wishlist')->getAddUrl($_product) ?>','wishlist','<?php echo $_product->getId() ?>');return false;" href="<?php echo $this->helper('wishlist')->getAddUrl($_product) ?>" ><?php echo $this->__('Add to Wishlist') ?></a></li>
                         <?php endif; ?>
                            <?php if($_compareUrl=$this->getAddToCompareUrl($_product)): ?>
                                <li><span class="separator">|</span> <a href="<?php echo $_compareUrl ?>" class="link-compare"><?php echo $this->__('Add to Compare') ?></a></li>
@@ -150,15 +160,21 @@
                 <?php endif; ?>
                    <div class="actions">
                        <?php if(!$_product->canConfigure() && $_product->isSaleable()): ?>
+                        <?php /* ?>
                         <button type="button" title="<?php echo $this->__('Add to Cart') ?>" class="button btn-cart" onclick="setLocation('<?php echo $this->getAddToCartUrl($_product) ?>')"><span><span><?php echo $this->__('Add to Cart') ?></span></span></button>
+                        <?php */ ?>
+                        <button type="button" title="<?php echo $this->__('Add to Cart') ?>" class="button btn-cart" onclick="aitunitsForm_productQty.send('<?php echo $this->getAddToCartUrl($_product) ?>','cart','<?php echo $_product->getId() ?>')"><span><span><?php echo $this->__('Add to Cart') ?></span></span></button>
                     <?php elseif($_product->getStockItem() && $_product->getStockItem()->getIsInStock()): ?>
                            <a title="<?php echo $this->__('View Details') ?>" class="button" href="<?php echo $_product->getProductUrl() ?>"><?php echo $this->__('View Details') ?></a>
                        <?php else: ?>
                            <p class="availability out-of-stock"><span><?php echo $this->__('Out of stock') ?></span></p>
                        <?php endif; ?>
                        <ul class="add-to-links">
                            <?php if ($this->helper('wishlist')->isAllow()) : ?>
+                            <?php /* ?>
                             <li><a href="<?php echo $this->helper('wishlist')->getAddUrl($_product) ?>" class="link-wishlist"><?php echo $this->__('Add to Wishlist') ?></a></li>
+                            <?php */ ?>
+                            <li><a class="link-wishlist" onclick="aitunitsForm_productQty.send('<?php echo $this->helper('wishlist')->getAddUrl($_product) ?>','wishlist','<?php echo $_product->getId() ?>');return false;" href="<?php echo $this->helper('wishlist')->getAddUrl($_product) ?>"><?php echo $this->__('Add to Wishlist') ?></a></li>
                         <?php endif; ?>
                            <?php if($_compareUrl=$this->getAddToCompareUrl($_product)): ?>
                                <li><span class="separator">|</span> <a href="<?php echo $_compareUrl ?>" class="link-compare"><?php echo $this->__('Add to Compare') ?></a></li>
